src/ifcdiff/ifcdiff.py

```diff
@@ -277,7 +277,14 @@ def diff_element_relationships(self, old, new):
             return
         for relationship in self.relationships:
             if relationship == "type":
-                if ifcopenshell.util.element.get_type(old) != ifcopenshell.util.element.get_type(new):
+                old_type = ifcopenshell.util.element.get_type(old)
+                new_type = ifcopenshell.util.element.get_type(new)
+                if old_type is not None and new_type is not None:
+                    if old_type.get_info() != new_type.get_info():
```

Comment:
I don't have a good overview of the entire chain of logic here, but `get_info()` will return a dict with values, which may still be ifcopenshell.,entity_instances. So if the type has an IfcOwnerHistory or RepresentationMaps and these are from different files, than the dictionary comparison will still evaluate to false. The dict also contains the numeric instance identifier. I don't know if it's intended to include these in the comparison. Consider: `recursive=True` for the first, `include_identifier=False` for the second.